libsolidity/interface/StandardCompiler.cpp

```diff
@@ -1313,15 +1315,32 @@ Json::Value StandardCompiler::compileSolidity(StandardCompiler::InputsAndSetting
 Json::Value StandardCompiler::compileYul(InputsAndSettings _inputsAndSettings)
 {
 	if (_inputsAndSettings.sources.size() != 1)
-		return formatFatalError("JSONError", "Yul mode only supports exactly one input file.");
+		return formatFatalError(
```

Comment:
The main thing is that I find it weird to forward `_inputsAndSettings.errors` here. Maybe it would be better to set a local flag, move the

`output["errors"] = std::move(_inputsAndSettings.errors);`

to the start of this function and do `output["errors"] += formatFatalError(...)` here.

Reply:
Why is it weird to forward the errors? After all `formatFatalError()` constructs full JSON output, just like `compileYul()` does so it makes sense to me that both would take over `_inputsAndSettings.errors` and add more stuff to it.

I can change it but I thought it was cleaner and more concise this way.

Reply:
I mean, I would understand the problem if `formatFatalError()` was returning a single error as the name implies but it does not.

Reply:
it's just that it is easy to forget to supply the right error list - so I'm wondering if it would not be better to just return an error and append it at the call site to `output["errors"]`.

Reply:
Another option would be to factor out this checking of fatal errors into its own function

Reply:
ok, so how about now. I now append to `output["errors"]` directly.